### PR TITLE
Fix for #27 Unwarranted circular property definition

### PR DESCRIPTION
**Pull request by psenechal-stingray**
...plus minor fix to avoid side effects on ExpansionBuffer::toString

### CODE REVIEW

**Comment by psenechal-stingray** [comment]
I'm not sure I understand correctly.
Do you mean that this isn't worth a PR (i.e. making an affirmation)? Feel free to close the PR then.
Or are you commenting on the fork, suggesting that I create a PR? If so, I already did: [here](https://github.com/mojohaus/properties-maven-plugin/pull/28).

**Comment by robertotru** [comment]
I meant that is was worth a PR. By the way, I had not time to review it for merging it in my branch. My bad, sorry!

File: src/test/java/org/codehaus/mojo/properties/PropertyResolverTest.java [hunk 1]
```diff
@@ -81,6 +81,19 @@ public void multipleValuesAreResolved()
     }
 
     @Test
+    public void propertyIncludesAnotherPropertyMoreThanOnce()
```
Comment (lpearson05) [inline src/test/java/org/codehaus/mojo/properties/PropertyResolverTest.java]: To match the previous method, wouldn't `multipleOccurrencesOfTheSameValueAreResolved ` be more appropriate?

**Comment by lpearson05** [comment]
@psenechal-stingray If I'm not mistaken, that link ...points to here. I'm confused.

Also, shouldn't there be a test for not a circular reference (e.g., `referencingAResolvedPropertyDoesNotThrow`), since ...the issue is that this exception is being thrown when we don't expect it.

File: src/test/java/org/codehaus/mojo/properties/PropertyResolverTest.java (hunk 1 above)
Comment (psenechal-stingray) [inline src/test/java/org/codehaus/mojo/properties/PropertyResolverTest.java]: One of my concerns naming this method was choosing a name that would be as descriptive as possible.

I agree that `multipleOccurencesOfTheSameValueAreResolved` is probably grammatically more akin to the other tests.

Then I would suggest `multipleOccurencesOfTheSamePropertyWithinAnotherAreResolved` or`multipleOccurencesOfTheSamePlaceholderWithinAnotherAreResolved` as the terms "property" and "placeholder" seem to be used interchangeably within the file, and the thing being resolved to the value is the property/placeholder.

It's hard to name things right, and I do not really have a strong opinion in this case. I'd let the final word on this to @robertotru.bertotru

**Comment by psenechal-stingray** [comment]
@lpearson05 I'm a bit confused too. As you've probably seen, I do not contribute frequently on GitHub. It looks like comments on the commit of my fork are also shown here, but not the other way around.

As for `referencingAResolvedPropertyDoesNotThrow`, isn't it already implicitly tested by the other tests? I cannot think of a test that would not be a duplicate of one or more other test. Please let me know if I'm missing something.

**Pull request merged**